### ISSUE TITLE
Not starting (Node 4.2.4, Ubuntu 15.10)

### ISSUE THREAD

**Issue by naholyr** [Jam3/devtool#2]
```sh
npm i -g devtool
devtool
```

Nothing happens :(

**Comment by mattdesl** [comment]
:cry: Thanks for testing. I can try putting some more errors in place.

Have you been able to run other Electron-based apps? Like [vmd](https://github.com/yoshuawuyts/vmd)?

```sh
npm i vmd -g
touch README.md
vmd README.md
```

**Comment by naholyr** [comment]
``vmd`` seems to be working fine. The difference may be in the version of Electron used:

For vmd:

> > electron-prebuilt@0.36.2 postinstall /home/nchambrier/.nvm/versions/node/v4.2.4/lib/node_modules/vmd/node_modules/electron-prebuilt

For devtool:

> > electron-prebuilt@0.35.6 postinstall /home/nchambrier/.nvm/versions/node/v4.2.4/lib/node_modules/devtool/node_modules/electron-prebuilt

I'm developing a Electron application myself which is based on 0.36.2 but worked fine with 0.35 few weeks ago so I'm not sure this is it. I'll try to upgrade dependency and check what happens.

**Comment by mattdesl** [comment]
It could be. I was noticing other errors with the latest Electron (https://github.com/atom/electron/issues/3832#issuecomment-172930036), which is why I'm sticking with 0.35 for `devtool` currently.

**Comment by verkholantsev** [comment]
Same problem on OS X 10.11.2 (15C50). After `devtool index.js` Electron icon appears in tray, but the app has no available windows.

**Comment by naholyr** [comment]
No difference with latest version

```sh
cd /path/to/devtool
npm i --save electron-prebuilt@latest
devtool # still nothing
```

Putting various console.log in ``server.js`` I'm not really sure what happens wrong. I can only say that [this line](https://github.com/Jam3/devtool/blob/master/server.js#L80) is reached twice (console.log shown twice) but I'll need time and/or hints to dig deeper ;)

**Comment by mattdesl** [comment]
You can try redirecting console and quitting on error:

```sh
devtool index.js -c -q
```

I will hunt around for a failing computer and see if I can reproduce.

**Comment by eth0lo** [comment]
Also happened to me, redirecting to console any console.log  appeared fine , also putting a debug statement breaks correctly, but not window is shown

**Comment by molokoloco** [comment]
Some error after simple instalation on Ubuntu
$ devtool
pci id for fd 15: 80ee:beef, driver (null)
OpenGL Warning: glFlushVertexArrayRangeNV not found in mesa table
OpenGL Warning: glVertexArrayRangeNV not found in mesa table
OpenGL Warning: glCombinerInputNV not found in mesa table
OpenGL Warning: glCombinerOutputNV not found in mesa table
OpenGL Warning: glCombinerParameterfNV not found in mesa table
OpenGL Warning: glCombinerParameterfvNV not found in mesa table
OpenGL Warning: glCombinerParameteriNV not found in mesa table
OpenGL Warning: glCombinerParameterivNV not found in mesa table
OpenGL Warning: glFinalCombinerInputNV not found in mesa table
OpenGL Warning: glGetCombinerInputParameterfvNV not found in mesa table
OpenGL Warning: glGetCombinerInputParameterivNV not found in mesa table
OpenGL Warning: glGetCombinerOutputParameterfvNV not found in mesa table
OpenGL Warning: glGetCombinerOutputParameterivNV not found in mesa table
OpenGL Warning: glGetFinalCombinerInputParameterfvNV not found in mesa table
OpenGL Warning: glGetFinalCombinerInputParameterivNV not found in mesa table
OpenGL Warning: glDeleteFencesNV not found in mesa table
OpenGL Warning: glFinishFenceNV not found in mesa table
OpenGL Warning: glGenFencesNV not found in mesa table
OpenGL Warning: glGetFenceivNV not found in mesa table
OpenGL Warning: glIsFenceNV not found in mesa table
OpenGL Warning: glSetFenceNV not found in mesa table
OpenGL Warning: glTestFenceNV not found in mesa table
libGL error: core dri or dri2 extension not found
libGL error: failed to load driver: vboxvideo
[8871:0120/170708:ERROR:buffer_manager.cc(342)] [GroupMarkerNotSet(crbug.com/242999)!:D06E021D8F330000]GL ERROR :GL_INVALID_ENUM : glBufferData: <- error from previous GL command
[8871:0120/170709:ERROR:texture_manager.cc(1995)] [GroupMarkerNotSet(crbug.com/242999)!:D062021D8F330000]GL ERROR :GL_INVALID_ENUM : glTexImage2D: <- error from previous GL command

**Comment by MCluck90** [comment]
Having the same issue on Windows 10 with Node v5.4.1. Running `devtool` launches 4 Electron processes but nothing is shown. Redirecting the console displays console logs correctly but did not break on a `debugger` statement. Showing the browser window `devtool -s` allows me to access a console tab and use as a REPL but, again, doesn't break on `debugger` statements.

**Comment by mattdesl** [comment]
Those experiencing trouble, please try the `-s` or `--show` flag to see if the DevTools opens:

```sh
devtool --show
```

It seems on some systems, Electron won't open the DevTools unless the main window is also shown. I wrote a patch which may fix this in `1.2.2`, but it will still show a very small window on some systems (Windows).

I'll put together a test case and open a new issue in Electron for this.

---

@MCluck90 You may need to reload (Cmd + R / Ctrl + R) after opening the window before break points / debugger starts working. (issue: https://github.com/atom/electron/issues/2379)

If that isn't working, please open a new issue. :smile:

---

@molokoloco It seems like you are having an unrelated issue.

**Comment by MCluck90** [comment]
@mattdesl That worked perfectly. Thanks! Also, thanks for making this. I'm sure it will come in handy. :sunglasses:

**Comment by ManRueda** [comment]
Another workaround is:
* Open devtool with -s or --show
* Detach the Developer Tool window from the main window (with the gear menu)
* Close it with window.close()

Now you can run devtool without -s or --show!! :sunglasses::sunglasses::sunglasses:

**Comment by mattdesl** [comment]
@ManRueda and others who are testing on windows, I just pushed another fix at `1.3.0` to try and improve the window hiding (especially for Windows where a tiny window might appear). Please let me know if it's any better/worse. :smile: 

```sh
# install latest
npm install devtool -g

# should show just DevTool REPL
devtool

# should show browser window + DevTools
devtool -s

# should show just browser window
devtool -s -h

# should not show any GUI
devtool -h
```

**Comment by MCluck90** [comment]
Windows 10 here. The workaround worked and now the update has fixed it properly. Thanks!

**Comment by ManRueda** [comment]
@mattdesl With Windows 7, everything works great now. Except for ```devtool -h```, shows the tiny windows. But i can live with that haha

**Comment by mattdesl** [comment]
Cool, I will close this and revisit the Windows issues in another ticket. Thanks! :+1:

**Comment by cutandpastey** [comment]
@mattdesl Has this issue been patched? Im experiencing it on ubuntu 14.04 with node 4. Im trying to run some mocha tests through CI so the workarounds suggested here don't fit my use case. Are there any other known solutions that I can run with `-h`?
